client/web/src/enterprise/batches/settings/CheckButton.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useRef } from 'react'
+
+import CheckIcon from 'mdi-react/CheckIcon'
+import CloseIcon from 'mdi-react/CloseIcon'
+
+import { Button, LoadingSpinner } from '@sourcegraph/wildcard'
+
+export interface CheckButtonProps {
+    label: string
+    onClick: React.MouseEventHandler
+    loading: boolean
+    successMessage?: string
+    failedMessage?: string
+}
+
+export const CheckButton: React.FunctionComponent<React.PropsWithChildren<CheckButtonProps>> = ({
+    label,
+    onClick,
+    loading,
+    successMessage,
+    failedMessage,
+}) => {
+    const buttonReference = useRef<HTMLButtonElement | null>(null)
+    return (
+        <>
+            {!loading && !successMessage && !failedMessage && (
+                <Button
+                    className="text-primary text-nowrap"
+                    onClick={onClick}
+                    variant="link"
+                    aria-label={label}
+                    ref={buttonReference}
+                >
+                    Check
+                </Button>
+            )}
+            {loading && (
+                <div className="text-white-50">
```

Comment:
`text-muted` might be a more appropriate class here, just to avoid any light/dark mode issues:

```suggestion
                <div className="text-muted">
```